src/Mailgun/Resource/Api/Domains/CredentialListResponse.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * Copyright (C) 2013-2016 Mailgun
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+namespace Mailgun\Resource\Api\Domains;
+
+use Mailgun\Resource\CreatableFromArray;
+
+/**
+ * @author Sean Johnson <sean@mailgun.com>
+ */
+class CredentialListResponse implements CreatableFromArray
+{
+    /**
+     * @var int
+     */
+    private $totalCount;
+
+    /**
+     * @var Credential[]
+     */
+    private $items;
+
+    /**
+     * @param array $data
+     *
+     * @return CredentialListResponse
+     */
+    public static function createFromArray(array $data)
+    {
+        $items = [];
+
+        foreach ($data['items'] as $item) {
+            $items[] = new Credential(
+                $item['size_bytes'],
+                new \DateTime($item['created_at']),
+                $item['mailbox'],
+                $item['login']
```

Comment:
Personally I like to check if the array has the expected key.

So maybe you can check - at least - if there is the `array_key_exists('items', $data)`. Also, given this is an array, provided as argument and we need to use these keys in order to build the `Credential` object, may be a good idea to check if the `$item` has all expected keys.

Is up to you what we need to do here: throw an exception or just ignore the invalid `$item` (if it does not contain all expected keys)